### PR TITLE
0/1D test for `BatchNorm` layer

### PR DESCRIPTION
**Pull request by Abdurrahheem**
This PR introduces support for 0/1D inputs in `BatchNorm` layer. 

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/dnn/src/layers/batch_norm_layer.cpp [hunk 1]
```diff
@@ -272,6 +277,26 @@ class BatchNormLayerImpl CV_FINAL : public BatchNormLayer
...
+            if (inputs[0].dims == 0) {  // Scalar input
+                float inpVal = inputs[0].at<float>();
+                float outVal = inpVal * weights_.at<float>(0) + bias_.at<float>(0);
+                outBlob.at<float>(0) = outVal;
+            } else {  // Vector input
+                for (int i = 0; i < inpBlob.size[0]; i++) {
+                    float inpVal = inpBlob.at<float>(i);
+                    float outVal = inpVal * weights_.at<float>(i) + bias_.at<float>(i);
+                    outBlob.at<float>(i) = outVal;
+                }
```
Comment (asmorkalov) [inline modules/dnn/src/layers/batch_norm_layer.cpp:293]: you can use cv::multiply and cv::add instead of loop. Or, even simpler outBlob = inpBlob * weights_ + bias_, if they have the same type and compatible shapes.

File: modules/dnn/test/test_layers_1d.cpp [hunk 2]
```diff
@@ -603,4 +603,52 @@ INSTANTIATE_TEST_CASE_P(/*nothting*/, Layer_FullyConnected_Test,
                             std::vector<int>({4})
 ));
 
+typedef testing::TestWithParam<tuple<std::vector<int>>> Layer_BatchNorm_Test;
```
Comment (asmorkalov) [inline modules/dnn/test/test_layers_1d.cpp:606]: `tuple` is redundant.

File: modules/dnn/test/test_layers_1d.cpp [hunk 3]
```diff
@@ -603,4 +603,52 @@ INSTANTIATE_TEST_CASE_P(/*nothting*/, Layer_FullyConnected_Test,
...
+TEST_P(Layer_BatchNorm_Test, Accuracy_01D)
+{
+    std::vector<int> input_shape = get<0>(GetParam());
+    bool has_weight = false;
+    bool has_bias = false;
```
Comment (asmorkalov) [inline modules/dnn/test/test_layers_1d.cpp:611]: variables are not used besides lp.set. It means that they may be removed.

File: modules/dnn/src/layers/batch_norm_layer.cpp [hunk 4]
```diff
@@ -272,6 +277,15 @@ class BatchNormLayerImpl CV_FINAL : public BatchNormLayer
...
         outputs_arr.getMatVector(outputs);
 
+        if (inputs[0].dims <= 1) { // Handling for 0D and 1D
+            Mat &inpBlob = inputs[0];
+            Mat &outBlob = outputs[0];
+            CV_Assert(inpBlob.total() == weights_.total());
+            cv::multiply(inpBlob, weights_, outBlob);
+            cv::add(outBlob, bias_, outBlob);
+            return;
+        }
```
Comment (dkurt) [inline modules/dnn/src/layers/batch_norm_layer.cpp:287]: Does OpenCL backend supports 1D?

File: modules/dnn/test/test_layers_1d.cpp [hunk 5]
```diff
@@ -603,4 +603,52 @@ INSTANTIATE_TEST_CASE_P(/*nothting*/, Layer_FullyConnected_Test,
...
+
+    // Layer parameters
+    LayerParams lp;
+    lp.type = "BatchNorm";
+    lp.name = "BatchNormLayer";
+    lp.set("has_weight", has_weight);
+    lp.set("has_bias", has_bias);
+
+    Mat meanMat(input_shape.size(), input_shape.data(), CV_32F, 0.5);
+    Mat varMat(input_shape.size(), input_shape.data(), CV_32F, 0.5);
```
Comment (dkurt) [inline modules/dnn/test/test_layers_1d.cpp:621]: randu is more prefererable

File: modules/dnn/src/layers/batch_norm_layer.cpp (hunk 4 above)
Comment (Abdurrahheem) [reply modules/dnn/src/layers/batch_norm_layer.cpp:287]: I do not think it does. On CI, I can not find any information related to OCV. I did not change anything in `#HAVE_OPENCL` part

File: modules/dnn/test/test_layers_1d.cpp (hunk 2 above)
Comment (Abdurrahheem) [reply modules/dnn/test/test_layers_1d.cpp]: done

File: modules/dnn/test/test_layers_1d.cpp (hunk 3 above)
Comment (Abdurrahheem) [reply modules/dnn/test/test_layers_1d.cpp]: done

File: modules/dnn/test/test_layers_1d.cpp (hunk 5 above)
Comment (Abdurrahheem) [reply modules/dnn/test/test_layers_1d.cpp]: Done

**Review by dkurt — APPROVED** [review]
